File: src/temperature_sensor.cpp
#include "ros/ros.h"
#include "sensor_msgs/Temperature.h"
#include <stdlib.h> //For rand()

int main(int argc, char **argv){
    ros::init(argc, argv, "temperature_sensor_cpp");
    ROS_INFO("Starting temperature_sensor_cpp...");
    ros::NodeHandle nodeHandle;
    ros::Publisher tempPub = nodeHandle.advertise<sensor_msgs::Temperature>("franek/temperature", 10);
    
    ros::Rate r(1); //1Hz
    ROS_INFO("Started temperature_sensor_cpp successfully. Publishing...");
    while(ros::ok()){
        //Generate a Temperature Message
        sensor_msgs::Temperature t;
        t.header.frame_id = "franek_link";
        t.header.stamp = ros::Time::now();
        t.temperature = rand() % 30; //Random Value between 0 an 30
        t.variance = 0; //0 = unknown

        tempPub.publish(t);
        ROS_INFO("Published Temp: %f", t.temperature);

        r.sleep();
    }
    
    ros::spin();   
}

File: scripts/temperature_sensor.py
#! /usr/bin/env python
import rospy, random, std_msgs.msg
from sensor_msgs.msg import Temperature

def publish_temp():
        pub = rospy.Publisher('henri/temperature', Temperature, queue_size=10)
        rospy.init_node('temperature_sensor_py', anonymous=True)
        rate = rospy.Rate(1)
        while not rospy.is_shutdown():
            t = Temperature()

            h = std_msgs.msg.Header()
            h.stamp = rospy.Time.now()
            h.frame_id = 'henri_link'

            t.header = h
            t.temperature = random.uniform(0.0, 30.0)
            t.variance = 0

            pub.publish(t)
            rospy.loginfo('Published Temp: %f', t.temperature)
            rate.sleep()

if __name__ == '__main__':
        try:
            publish_temp()
        except rospy.ROSInterruptException:
            pass
